quil-rs/src/expression/simplification.rs

```diff
@@ -462,43 +463,58 @@ static RULES: Lazy<Vec<Rewrite>> = Lazy::new(|| {
         // Largely copied from https://github.com/egraphs-good/egg/blob/82c00e970f0bc1fbfe90ce6dc3c3c79ee919c933/tests/math.rs
         // and https://github.com/herbie-fp/herbie/blob/2052806f2ffe0d46bc2e151dd096b127c39e12bd/egg-herbie/src/rules.rs
 
+        // pos and neg
+        rw!("pos canon"     ; "(pos ?a)"                => "?a"),
```

Comment:
What do you think about removing `pos` (and perhaps `-` subtraction) in the name of simplifying both the e-graphs and the ruleset?

Reply:
I'd love to remove the prefix unary ops if we could, but they're part of the Quil spec, right? Either way, I think these `<X> canon` rules fire almost only at the first iteration of simplification, from anecdotally looking at `RUST_LOG=egg=debug` runs of things, and then both `pos` and `neg` are pretty much never heard from again—barring when something simplifies a great deal.

Reply:
Oh! But if we change how `Expression`s get turned into `RecExpr<Expr>`s, we of course could simplify the `Expr` language, couldn't we? That'd be nice.